src/S3/S3Client.php

```diff
@@ -1035,6 +1056,29 @@ public static function applyDocFilters(array $api, array $docs)
             'shapes' => ['PutObjectRequest', 'UploadPartRequest']
         ];
 
+        // Add `ExpiresString` shape to output structures which contain `Expires`
```

Comment:
Something we could consider, maybe later, is to split the implementation of this function in sub-functions for each docs injected here. For example, since you are adding a docs filter for Expires and ExpiresString you can create a specific method for that and just make the call from this general method. By doing this we can keep the code cleaner and more easy to maintain.

Reply:
You make a good point.  If we get the time at some point in the future, I'd like to move this logic off of service clients entirely— it doesn't belong here.  Which is why, for now, I won't add sub-methods for this.  I think one method dedicated to docs customizations is enough of an eyesore for now 😜